src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs

```diff
@@ -57,18 +66,33 @@ public Uri SourceUri
             {
                 var requestTimeout = TimeSpan.FromSeconds(timeoutInSecond);
                 tokenSource.CancelAfter(requestTimeout);
-
+                bool packageUseTempApiKey = false;
+                bool symbolUseTempApiKey = false;
                 var apiKey = getApiKey(_source);
 
-                await PushPackage(packagePath, _source, apiKey, requestTimeout, log, tokenSource.Token);
+                // If user push to https://nuget.smbsrc.net/ directly, use temp api key.
+                if (_source.Equals(NuGetConstants.DefaultSymbolServerUrl, StringComparison.OrdinalIgnoreCase))
```

Comment:
@zhili1208 mentioned there are some other scenarios/paths that this shouldn't be used for